src/utils/privateRoute.tsx

```diff
@@ -0,0 +1,15 @@
+import { ROUTE_PATH } from '../routes';
+import { setUser } from '../store/user/slice';
+import { useAppDispatch } from '../store/hooks';
+import { Navigate, Outlet } from 'react-router-dom';
+
+const PrivateRoute = () => {
+  const dispatch = useAppDispatch();
```

Comment:
Let's create a custom hook, to handle actions in Private in Public routes. `useUserAuth` hook or smthg like that

Reply:
fix: moved to custom useUserAuth  hook